### ISSUE TITLE
flutter-gold check breaks while traversing PRs

### ISSUE THREAD

**Issue by Piinks** [flutter/flutter#65979]
The flutter-gold check appears to be broken.
The check runs as a cron job every five minutes, during which it iterates through all open PRs to update the check status.

According to the cron job listing, it has failed, but I cannot find an error log for the failure: [here](https://pantheon.corp.google.com/appengine/cronjobs?project=flutter-dashboard)

It appears from the gold-status logs that it fails when it reaches #62502, and the rest of the PRs aren't checked: [here](https://pantheon.corp.google.com/logs/viewer?project=flutter-dashboard&resource=gae_app%2Fmodule_id%2Fdefault%2Fversion_id%2Fversion-939ebe3&minLogLevel=0&expandAll=false&timestamp=2020-09-16T22:23:33.129000000Z&customFacets=&limitCustomFacetWidth=true&dateRangeStart=2020-09-16T21:23:39.817Z&dateRangeEnd=2020-09-16T22:23:39.817Z&interval=PT1H&logName=projects%2Fflutter-dashboard%2Flogs%2Fappengine.googleapis.com%252Fstdout&logName=projects%2Fflutter-dashboard%2Flogs%2Fappengine.googleapis.com%252Fnginx.request&logName=projects%2Fflutter-dashboard%2Flogs%2Fappengine.googleapis.com%252Frequest_log&scrollTimestamp=2020-09-16T22:22:47.701000000Z&filters=text:gold-status)

@kjlubick could this be caused by the new versioned endpoint? This check hits `http://flutter-gold.skia.org/json/changelist/github/${pr.number}/${pr.head.sha}/untriaged`

**Comment by godofredoc** [comment]
There are two problems with the endpoint:

* It is making a request to a url that doesn't exist and returning an http status code 400.
* The endpoint is not managing the 400 not found status code properly as it is not showing up as an error.

**Comment by kjlubick** [comment]
Ah, I know why https://flutter-gold.skia.org/json/changelist/github/62502/64c44092fc8c69db6501ebc82db7ae43410c044d/untriaged is returning 500. I recently had to make a change where for older PR, we no longer fetch data to fulfil that request. It was getting too expensive to return the data except when it was cached. A 500 error is potentially problematic, so I then changed it to return nothing.

If a PR has not had any activity for more than 20 days (i.e. no tryjobs run in 20 days that produced Gold data) or is closed, that API will not return accurate data any more. I can deploy the change that makes it not be a 500 error today, if you like. I cannot rollback the other part of the change, although in the next quarter or 2 as I rework the data backend, I'll probably be able to restore that API.
